internal/services/hybridkubernetes/hybrid_kubernetes_connected_cluster_resource_test.go

```diff
@@ -0,0 +1,193 @@
+	template := r.template(data)
+	return fmt.Sprintf(`
+				%s
+
+resource "azurerm_hybrid_kubernetes_connected_cluster" "test" {
+  name                         = "acctest-k-%d"
+  resource_group_name          = azurerm_resource_group.test.name
+  agent_public_key_certificate = "MIICCgKCAgEAsSpALlON3394ysLQdRSy6cCBwL08NgZp7c1xsy0kQH/wHuixfoCwtL1OZ0a5kqod9vE6L8ICsXAE+iEdU1OspcJxL9J/gSyiOCMYPUabbYRXFy5x258RRLtn60NoaqcaDW+Z80HLwJOMECdJ/yDkuuNbnL0J2cyR8/WXjoeee8cG52QmDuxB6a4ROOushroIE2NS3FuhJh3b3Ddj+NU3gfbgIjjCMJQDpJWqUi/68lB33K1FQwuY1CCT8rKC5dDRKnlTgEWtnhBBly1D+C9GPTh7l9NTxcAEcBKo1ZIkMcixcS+gTcsTtDRRwuTtt8kybohfRMvmBA/S9bkD6cCaGJMe8YerOyJsDd4zSUHz/qN9iTt8FDdFcCIjhEdsMKl350wMj7+UNPvchlRAAve3oCIntW9063fDlQHrsaPfhCplTlKLAt1jcUkGSaeuylnRi8te+hmDCIhmo8wDqMv8Yvy7BTX4bYMg+6j0EGiIKPMRf5NHt6bXbiWEcO5LWEwgCbvpTjf7XdrU/xJ+eB+uAP1etvE0tCYAEFisfFnqNZxMQCFjovI0ZNxizrZOBznk15fWTr4KTNsPUTkEvbJfZPQqc/QyC5yqIAFHAAj+jusd4tK9f19nCsoi78xeKGH2s1zrD0AmBNGUrPLPVzgfdueadJKbVbiPteYec9qSG50CAwEAAQ=="
+  identity {
+    type = "SystemAssigned"
+  }
+  location = "%s"
+}
+`, template, data.RandomInteger, data.Locations.Primary)
+}
+
+func (r HybridKubernetesConnectedClusterResource) requiresImport(data acceptance.TestData) string {
+	config := r.basic(data)
+	return fmt.Sprintf(`
+			%s
+
+resource "azurerm_hybrid_kubernetes_connected_cluster" "import" {
+  name                         = azurerm_hybrid_kubernetes_connected_cluster.test.name
+  resource_group_name          = azurerm_resource_group.test.name
+  agent_public_key_certificate = azurerm_hybrid_kubernetes_connected_cluster.test.agent_public_key_certificate
+  identity {
+    type = "SystemAssigned"
+  }
+  location = azurerm_hybrid_kubernetes_connected_cluster.test.location
+}
+`, config)
+}
+
+func (r HybridKubernetesConnectedClusterResource) complete(data acceptance.TestData) string {
+	template := r.template(data)
+	return fmt.Sprintf(`
+			%s
+
+resource "azurerm_hybrid_kubernetes_connected_cluster" "test" {
+  name                         = "acctest-k-%d"
+  resource_group_name          = azurerm_resource_group.test.name
+  agent_public_key_certificate = "MIICCgKCAgEAsSpALlON3394ysLQdRSy6cCBwL08NgZp7c1xsy0kQH/wHuixfoCwtL1OZ0a5kqod9vE6L8ICsXAE+iEdU1OspcJxL9J/gSyiOCMYPUabbYRXFy5x258RRLtn60NoaqcaDW+Z80HLwJOMECdJ/yDkuuNbnL0J2cyR8/WXjoeee8cG52QmDuxB6a4ROOushroIE2NS3FuhJh3b3Ddj+NU3gfbgIjjCMJQDpJWqUi/68lB33K1FQwuY1CCT8rKC5dDRKnlTgEWtnhBBly1D+C9GPTh7l9NTxcAEcBKo1ZIkMcixcS+gTcsTtDRRwuTtt8kybohfRMvmBA/S9bkD6cCaGJMe8YerOyJsDd4zSUHz/qN9iTt8FDdFcCIjhEdsMKl350wMj7+UNPvchlRAAve3oCIntW9063fDlQHrsaPfhCplTlKLAt1jcUkGSaeuylnRi8te+hmDCIhmo8wDqMv8Yvy7BTX4bYMg+6j0EGiIKPMRf5NHt6bXbiWEcO5LWEwgCbvpTjf7XdrU/xJ+eB+uAP1etvE0tCYAEFisfFnqNZxMQCFjovI0ZNxizrZOBznk15fWTr4KTNsPUTkEvbJfZPQqc/QyC5yqIAFHAAj+jusd4tK9f19nCsoi78xeKGH2s1zrD0AmBNGUrPLPVzgfdueadJKbVbiPteYec9qSG50CAwEAAQ=="
+  distribution                 = "kind"
+
+  identity {
+    type = "SystemAssigned"
+  }
+  infrastructure = "generic"
```

Comment:
are either of these are valid values?

Reply:
Yes, I've test it against a Kubernetes deployed on Kind following this guidance(https://docs.microsoft.com/en-us/azure/azure-arc/kubernetes/quickstart-connect-cluster?tabs=azure-cli).